spec/services/work_packages/set_attributes_service_spec.rb

```diff
@@ -1058,6 +1058,46 @@
       end
     end
 
+    context "with zero duration" do
+      let(:work_package) do
+        build_stubbed(:work_package, start_date: Time.zone.today, due_date: Time.zone.today + 5.days)
+      end
+      let(:call_attributes) { { duration: 0 } }
+      let(:expected_attributes) { {} }
+
+      it_behaves_like "service call" do
+        it "keeps the dates and duration values (error to be detected by contract)" do
+          subject
+
+          expect(work_package.start_date)
+            .to eq(Time.zone.today)
+          expect(work_package.due_date)
+            .to eq(Time.zone.today + 5.days)
+          expect(work_package.duration)
+            .to eq(0)
+        end
+      end
+
+      context "when the work package has a soonest_start from a predecessor (Regression #63598)" do
+        before do
+          allow(instance).to receive(:new_start_date).and_return(Time.zone.yesterday)
+        end
+
+        it_behaves_like "service call" do
+          it "keeps the dates and duration values (error to be detected by contract)" do
+            subject
+
+            expect(work_package.start_date)
+              .to eq(Time.zone.yesterday)
+            expect(work_package.due_date)
+              .to be_nil
```

Comment:
This currently assigns `nil` to the due_date. We could arguably return the previous due_date instead? It does not really matter, as the contract rejects the change after all.

Reply:
It's true that it does not really matter as the contract rejects the change, but on the modal, the end date disappears which feels wrong. Besides, on a manually scheduled work package, the end date would remain if the duration is 0, as demonstrated by these tests:

https://github.com/opf/openproject/blob/083f939fe35ce88233405cbdd8a2349f26b05289/spec/services/work_packages/set_attributes_service_spec.rb#L1040-L1082

and this screenshot:

![image](https://github.com/user-attachments/assets/ab7120cb-f480-4b18-957a-1aca40eff201)

So I think it would be more consistent to keep the due date here too.
I'll fix it.